llvm/lib/Target/SPIRV/SPIRVInstructionSelector.cpp

```diff
@@ -2527,6 +2640,11 @@ bool SPIRVInstructionSelector::selectIntrinsic(Register ResVReg,
   case Intrinsic::spv_udot:
   case Intrinsic::spv_sdot:
     return selectIntegerDot(ResVReg, ResType, I);
+  case Intrinsic::spv_dot4add_i8packed:
+    if (STI.canUseExtension(SPIRV::Extension::SPV_KHR_integer_dot_product) ||
```

Comment:
Nice to have this option. Now that we do, it would be nice to update the regular dot product to use this in the integer cases, but that can be left for a different change.

Reply:
Created for tracking here: https://github.com/llvm/llvm-project/issues/114632